ibis/sql/postgres/compiler.py

```diff
@@ -536,6 +536,43 @@ def _table_column(t, expr):
     return out_expr
 
 
+def _round(t, expr):
+    arg, digits = expr.op().args
+    sa_arg = t.translate(arg)
+
+    if digits is None:
+        return sa.func.round(sa_arg)
+
+    # postgres doesn't allow rounding of double precision values to a specific
+    # number of digits (though simple truncation on doubles is allowed) so
+    # we cast to numeric and then cast back if necessary
+    result = sa.func.round(sa.cast(sa_arg, sa.NUMERIC), t.translate(digits))
+    if digits is not None and isinstance(arg.type(), dt.Decimal):
+        return result
+    return sa.cast(result, sa.dialects.postgresql.DOUBLE_PRECISION())
+
+
+def _mod(t, expr):
+    left, right = map(t.translate, expr.op().args)
+
+    # postgres doesn't allow modulus of double precision values, so upcast and
+    # then downcast later if necessary
+    if not isinstance(expr.type(), dt.Integer):
+        left = sa.cast(left, sa.NUMERIC)
+        right = sa.cast(right, sa.NUMERIC)
+
+    result = left % right
+    if expr.type().equals(dt.double):
+        return sa.cast(result, sa.dialects.postgresql.DOUBLE_PRECISION())
+    else:
+        return result
+
+
+def _floor_divide(t, expr):
+    left, right = map(t.translate, expr.op().args)
+    return sa.func.floor(left / right)
```

Comment:
No, it yields integers. This is implemented so that it works regardless of the type of `left` and `right`.